### PR TITLE
add basic project template

### PR DESCRIPTION
**Pull request by cosmicBboy**
This PR:
- creates a `template/basic` cookiecutter template for general contributors
- update the `README.md` with instructions on how to set up a new project (with special notes for hackathon participants)

### Review Plan

@samhita-alla The README.md instructions are almost complete, note that the instructions for running the streamlit app against the `playground` cluster is currently not working.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,252 @@
...
+
+> **NOTE:** Make sure you're using `Python > 3.7`
+
+We use `cookiecutter` to manage project templates.
+
+> **NOTE:** For [MLOps Community Engineering Labs Hackathon](https://flyte.org/hackathon/) participants:
+> Each team will have its own branch on the main `flyteorg/flytelab` repo. If you're part of a team of more
+> than one person, assign one teammate to create a project directory and push it into your team's branch.
+
+Install prerequisites:
```
Comment (samhita-alla) [inline README.md:34]: Shall we talk about creating a branch and virtualenv? The note talks about creating a branch, but I believe it'd be clear to include that as part of the prerequisites.

Moreover, if the user isn't a committer, should we mention that "fork" is how the user has to start working on a Flytelab project?

File: projects/my_project/.dockerignore [hunk 2]
```diff
@@ -0,0 +1 @@
+!.flyte
```
Comment (samhita-alla) [inline projects/my_project/.dockerignore:1]: Do we want to have `my_project` or is this directory committed by mistake?

File: README.md [hunk 3]
```diff
@@ -1,3 +1,252 @@
...
+
+```bash
+cd projects/my_project
+
+# create and activate virtual environment
+python -m venv env
+source env/bin/activate
+
+# install requirements
+pip install -r requirements.txt -r requirements-dev.txt
```
Comment (samhita-alla) [inline README.md:84]: Do we need the libraries included in the `requirements-dev.txt` file?

File: README.md [hunk 4]
```diff
@@ -1,3 +1,252 @@
...
+</details>
+
+### Sandbox Deployment
+
+Start the sandbox cluster:
+
+```bash
+flytectl sandbox start --source .
+```
+
```
Comment (samhita-alla) [inline README.md:143]: Do we want to mention the troubleshooting guide here?

File: README.md [hunk 5]
```diff
@@ -1,3 +1,252 @@
...
+You should now be able to go to `http://localhost:30081/console` on your browser to see the Flyte UI.
+
```
Comment (samhita-alla) [inline README.md:145]: Do we have to ask the user to commit changes to a PR before running `python deploy.py`?

File: templates/basic/{{cookiecutter.project_name}}/deploy.py [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    for line in docker_client.api.push(image.tags[0], stream=True, decode=True):
+        typer.echo(line)
+
+
+def serialize(tag: str):
+    typer.echo("Serializing Flyte workflows")
+    subprocess.run([
+        "pyflyte",
+        "-c", "flyte.config",
+        "--pkgs", "my_project",
```
Comment (samhita-alla) [inline templates/basic/{{cookiecutter.project_name}}/deploy.py:109]:
```suggestion
        "--pkgs", "{{cookiecutter.project_name}}",
```

File: README.md [hunk 7]
```diff
@@ -1,3 +1,252 @@
...
+### Union.ai Playground Deployment
+
+The [Union.ai](https://union.ai/) team maintains a playground Flyte cluster that you can use
+to run your workflows.
+
+When you're ready to deploy to a full-fledged production Flyte cluster, first you'll need to
+get an account. Currenty, you'll need to reach out to the Union.ai team on the Flyte OSS Slack 
+[`#flytelab` channel](https://flyte-org.slack.com/archives/C032ZU3FSAX) to request an account,
+we're you'll receive a login, password, and in addition to that a `client_id` and `client_secret`
```
Comment (samhita-alla) [inline README.md:193]:
```suggestion
where you'll receive a login, password, and in addition to `client_id` and `client_secret`
```

File: README.md (hunk 1 above)
Comment (cosmicBboy) [reply README.md:34]: > Shall we talk about creating a branch and virtualenv

the environment that creates the project is different from the environment that runs the project workflows, so I didn't want to bog down/confuse newcomers with having to create a virtual environment just to install `cookiecutter` on it, then creating _another_ virtual environment for the project. I think this is a nice-to-have but not really necessary.

> The note talks about creating a branch, but I believe it'd be clear to include that as part of the prerequisites

👍🏽

> Moreover, if the user isn't a committer, should we mention that "fork" is how the user has to get started?

cool, will add

**Review by samhita-alla — APPROVED** [review]

**Pull request merged**